evm/vm/message.py

```diff
@@ -142,3 +138,45 @@ def storage_address(self, value):
     @property
     def is_create(self):
         return self.to == CREATE_CONTRACT_ADDRESS
+
+
+class ShardingMessage(Message):
+
+    def __init__(self,
+                 gas,
+                 gas_price,
+                 to,
+                 sender,
+                 value,
+                 data,
+                 code,
+                 origin=None,
+                 access_list=None,
+                 depth=0,
+                 create_address=None,
+                 code_address=None,
+                 should_transfer_value=True,
+                 is_static=False):
+        super(ShardingMessage, self).__init__(
+            gas=gas,
+            gas_price=gas_price,
+            to=to,
+            sender=sender,
+            value=value,
+            data=data,
+            code=code,
+            origin=origin,
+            depth=depth,
+            create_address=create_address,
+            code_address=code_address,
+            should_transfer_value=should_transfer_value,
+            is_static=is_static,
+        )
+
+        if access_list is not None:
+            validate_access_list(access_list)
+        self.access_list = access_list
+
+    @property
+    def is_create(self):
+        return self.code not in (None, b'')
```

Comment:
Hrm, it'd be nice to have a single representation of *empty*.  What do you think about changing `message.code` to always be a bytestring so that `b''` is the only *empty* value and we can just `return bool(self.code)`